### ISSUE TITLE
KeyError Exception Occurred

### ISSUE THREAD

**Issue by Cmdrd** [Lienus10/yoptions#7]
In the example query the following error is thrown:

```
File main.py", line 6, in <module>
    chain = yo.get_chain_greeks(stock_ticker='F', dividend_yield=0, option_type='c', risk_free_rate=None)
KeyError: -1

```

**Comment by NewTrail-120** [comment]
Also had this issue on 3/1/23 and 3/2/23 -  the XML URL does not provide a parsed List of Lists when it is the first week of the month or the first day of the week.  Thus the index(s) [-1] does not exist.  The data is there but there is no index to get the data as defined.

M update was to use the CSV file and parse as below :

file = urllib.request.urlopen("https://home.treasury.gov/resource-center/data-chart-center/interest-rates/daily-treasury-rates.csv/all/202303?type=daily_treasury_yield_curve&amp;field_tdr_date_value_month=202303&amp;page&amp;_format=csv")

    data = file.read()
    file.close()

    ts = str(data).split('\\n')[1].split(',')

    maturity = ['BC_DATE',
                'BC_1MONTH',
                'BC_2MONTH',
                'BC_3MONTH',
                'BC_4MONTH',
                'BC_6MONTH',
                'BC_1YEAR',
                'BC_2YEAR',
                'BC_3YEAR']

    data ={maturity: ts for maturity , ts in zip(maturity, ts)}

Still need to pass in the month and year above  202303 - but good for this month.
The CSV option adds the new data to the top of the file, so the current date's data is at the first index.

your mileage may vary......

**Comment by NewTrail-120** [comment]
Follow-up:
The above is in the  __risk_free function.  Also the rates need to be divided by 100 to change from % back to decimal for use the calculations.

Added the year and month to the query using the imported datetime library to generate the URL for the csv file.

    url_csv = "https://home.treasury.gov/resource-center/data-chart-center/interest-rates/daily-treasury-rates.csv/all/" + \
                datetime.date.strftime(datetime.datetime.now(),"%Y%m") +  \
                "?type=daily_treasury_yield_curve&amp;field_tdr_date_value_month=" + \
                    datetime.date.strftime(datetime.datetime.now(),"%Y%m") + "&amp;page&amp;_format=csv"

**Comment by Lienus10** [comment]
The problem should be resolved with the latest version 0.0.4

Would be nice, if you could confirm that it works now

**Comment by Cmdrd** [comment]
Update appears to have resolved the issue, thank you for the quick turnaround!
